src/cmd/update.rs

```diff
@@ -0,0 +1,94 @@
+use std::env;
+use std::fs;
+use std::fs::Permissions;
+use std::os::unix::prelude::*;
+use std::path::Path;
+
+use crate::tempfile::Tempfile;
+use anyhow::{anyhow, Context, Result};
+use os_info::{Info, Type};
+use serde::Deserialize;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const LATEST_RELEASE_URL: &str = "https://api.github.com/repos/sbstp/kubie/releases/latest";
+const FILENAME: &str = "kubie";
+
+#[derive(Debug, Deserialize)]
+pub struct Release {
+    tag_name: String,
+    prerelease: bool,
+    assets: Vec<Asset>,
+}
+
+#[derive(Debug, Deserialize)]
+struct Asset {
+    name: String,
+    browser_download_url: String,
+    state: String,
+}
+
+pub fn update() -> Result<()> {
+    let latest_release: Release = get_latest_release()?;
+    let latest_version = latest_release.tag_name;
+    if latest_version.eq(&format!("v{}", VERSION)) {
+        println!("Kubie is up-to-date : v{}", VERSION);
+    } else {
+        println!(
+            "A new version of Kubie is available ({}), the new version will be automatically installed...",
+            latest_version
+        );
+        let mut linux_download_url = String::new();
```

Comment:
This is not really idiomatic Rust. I think it would be cleaner if you created methods on the `Release` type, something like `get_linux_binary_url(&self) -> Option<&str>` that can be called when we match on the os type. If the URL is not found we can return an error.

```rust
match os_info::get().os_type() {
  os_info::Type::Macos => {
    download_url = latest_release.get_linux_binary_url().ok_or_else(|| anyhow!("..."))?;
```
The `ok_or_else` function allows you to map over None and return it into a Result. It essentially does:

```rust
match latest_release.get_linux_binary_url() {
  Some(x) => Ok(x),
  None => Err(anyhow!("....")),
}
```

Reply:
I did some refactoring in https://github.com/sbstp/kubie/pull/9/commits/3034def9819786bb0c7ffbd437da31124fe55d96 for this